### PR TITLE
Mimes can no longer break their vow.

### PR DESCRIPTION
**Pull request by Iamgoofball**
You sign up for mime, you carry it as a badge of honor.

### CODE REVIEW

**Comment by Qustinnus** [comment]
just add a debuff for when they break vow instead

**Comment by Cyberboss** [comment]
Traitor/ling mimes should still get this

**Comment by Iamgoofball** [comment]
@Cyberboss are they datumized? if so I'll do that

**Comment by praisenarsie** [comment]
@Iamgoofball traitors and lings are datumized if that's what you are asking

https://github.com/tgstation/tgstation/tree/master/code/datums/antagonists

**Comment by GuppyLaxx** [comment]
Cult mimes

**Comment by Iamgoofball** [comment]
@GuppyLaxx sounds like a personal problem

**Comment by ExcessiveUseOfCobblestone** [comment]
only for cc, bloodcult actually lets them invoke silently iirc.

**Comment by 81Denton** [comment]
@Qustinnus spawn a ghost role silent assassin if they break their vow

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
Give them the spell instead of forcing them off the vow. If it would bootyblast you that it can be toggled change it to one way, but at least it's their option when it comes off so they can use the wall spell until then.

I also think the mime's traitor spells require him to have this.

**Comment by Iamgoofball** [comment]
No.

**Comment by ExcessiveUseOfCobblestone** [comment]
Ugh reruns!

File: code/datums/antagonists/datum_traitor.dm [hunk 1]
```diff
@@ -41,12 +41,22 @@
 			if(!silent)
 				to_chat(traitor_mob, "Your training has allowed you to overcome your clownish nature, allowing you to wield weapons without harming yourself.")
 			traitor_mob.dna.remove_mutation(CLOWNMUT)
+	if(owner.assigned_role == "Mime")
+		var/mob/living/carbon/human/traitor_mob = owner.current
+		if(traitor_mob && istype(traitor_mob))
+			if(!silent)
+				to_chat(traitor_mob, "Your vow of silence is not more important than your goals, therefore it has been lifted.")
```
Comment (nicbn) [inline code/datums/antagonists/datum_traitor.dm]: You can check if they are traitors by making them scream. Please make it toggleable

**Review by nicbn — CHANGES_REQUESTED** [review]

**Comment by Armhulen** [comment]
we shouldn't let traitors talk a single a mime if you pick the job with a disadvantage you should also be ready to play with that disadvantage traitor or not, no going around it

File: code/datums/antagonists/datum_traitor.dm (hunk 1 above)
Comment (Iamgoofball) [reply code/datums/antagonists/datum_traitor.dm]: A true mime refuses the scream.

I can make traitor mimes fake not screaming.

**Comment by AdamElTablawy** [comment]
@Armhulen 

so we give traitor clowns clumsy then

**Comment by DaxDupont** [comment]
@Armhulen by that logic we should no longer remove the clusmy gene on traitor cluwne

**Comment by MrDoomBringer** [comment]
@Iamgoofball theres you should remove that tip of the round somewhere that mentions mimes breaking their vow.

**Comment by Iamgoofball** [comment]
Will do

**Comment by Incoming5643** [comment]
Absolutely nothing should be attached to assigned_role outside of antag things. It can't be changed during the course of a round by any means and creates an arbitrary boundary between one job and everyone else.

Job Intrinsics are bad (see #26869)

**Comment by Bawhoppen** [comment]
As much as I think we shouldn't have mimes ever breaking their vow, you have to remember the point of view where not everything should be outstandingly wacky. Make it a config option probably...

**Comment by PsyCommando** [comment]
What about spawning mimes missing their tongue organ? You could even work that into the lore.

**Comment by Cyberboss** [comment]
gooooof

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
If you're forcing the vow off as traitor, remove
https://github.com/tgstation/tgstation/blob/65bb0a3e32a5e8dfc22f7c9ea2df8312b4215455/code/modules/spells/spell_types/mime.dm#L83
and
https://github.com/tgstation/tgstation/blob/65bb0a3e32a5e8dfc22f7c9ea2df8312b4215455/code/modules/spells/spell_types/mime.dm#L119

**Comment by Iamgoofball** [comment]
@Incoming5643 we just split up R&D because of that thinking of "everyone should be able to do everything"

**Comment by KorPhaeron** [comment]
its a stale conflicting goof PR and we know how those end

**Pull request closed without merging**